web_src/js/components/RepoContributors.vue

```diff
@@ -0,0 +1,249 @@
+            </p>
+          </div>
+        </div>
+        <div class="ui attached segment">
+          <Line
+            :data="toGraphData(contributor.weeks, 'contributor')"
+            :options="getOptions('contributor')"
+          />
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import {createApp} from 'vue';
+import {
+  Chart,
+  Title,
+  Tooltip,
+  Legend,
+  BarElement,
+  CategoryScale,
+  LinearScale,
+  TimeScale,
+  PointElement,
+  LineElement,
+  Filler,
+} from 'chart.js';
+import zoomPlugin from 'chartjs-plugin-zoom';
+import {Line} from 'vue-chartjs';
+import 'chartjs-adapter-dayjs-4/dist/chartjs-adapter-dayjs-4.esm';
+
+const {pageData} = window.config;
+
+Chart.register(
+  TimeScale,
+  CategoryScale,
+  LinearScale,
+  BarElement,
+  Title,
+  Tooltip,
+  Legend,
+  PointElement,
+  LineElement,
+  Filler,
+  zoomPlugin
+);
+
+const sfc = {
+  components: {Line},
+  data: () => {
+    return {
+      isLoading: false,
+      totalStats: {},
+      repoLink: pageData.repoLink || [],
+      type: pageData.contributionType,
+      contributorsStats: [],
+    };
+  },
+  mounted() {
+    this.fetchGraphData();
+  },
+  computed: {
+    sortedContributors() {
+      return Object.values(this.contributorsStats)
+        .sort((a, b) =>
+          a.total_commits > b.total_commits ?
+            -1 :
+            a.total_commits === b.total_commits ?
+              0 :
+              1
+        )
+        .slice(0, 100);
+    },
+  },
+  methods: {
+    async fetchGraphData() {
+      this.isLoading = true;
+      fetch(`${this.repoLink}/contributors/data`)
+        .then((response) => response.json())
+        .then((data) => {
+          const {Total, ...rest} = data;
+          this.contributorsStats = rest;
+          this.totalStats = Total;
+          this.isLoading = false;
+        });
+    },
+    maxMainGraph() {
+      const maxValue = Math.max(
+        ...this.totalStats.weeks.map((o) => o[this.type])
+      );
+      const [cooefficient, exp] = maxValue
+        .toExponential()
+        .split('e')
+        .map(Number);
+      if (cooefficient % 1 === 0) {
+        return maxValue;
+      }
+      return (1 - (cooefficient % 1)) * 10 ** exp + maxValue;
+    },
+    additions(data) {
+      return Object.values(data).reduce((acc, item) => {
+        return acc + item.additions;
+      }, 0);
+    },
+    deletions(data) {
+      return Object.values(data).reduce((acc, item) => {
+        return acc + item.deletions;
+      }, 0);
+    },
+
+    toGraphData(data, type) {
+      return {
+        datasets: [
+          {
+            data: data.map((i) => {
+              return {x: i.week, y: i[this.type]};
+            }),
+            pointRadius: 0,
+            pointHitRadius: 0,
+            fill: 'start',
+            backgroundColor:
+              type === 'main' ?
+                'rgba(137, 191, 154, 0.6)' :
+                'rgba(96, 153, 38, 0.7)',
```

Comment:
No hardcoded colors please, use our variables, e.g. `var(--primary-color)` etc:

https://github.com/go-gitea/gitea/blob/main/web_src/css/base.css#L1

Reply:
Done. Now the graph color also changes based on contribution type. 
- commits: blue
- additions: green
- deletions: red

Reply:
Sounds good, I will test it soon.